pkg/apis/core/v1/helper/helpers.go

```diff
@@ -315,34 +316,43 @@ func NodeSelectorRequirementKeysExistInNodeSelectorTerms(reqs []v1.NodeSelectorR
 // MatchNodeSelectorTerms checks whether the node labels and fields match node selector terms in ORed;
 // nil or empty term matches no objects.
 func MatchNodeSelectorTerms(
+	node *v1.Node,
 	nodeSelectorTerms []v1.NodeSelectorTerm,
-	nodeLabels labels.Set,
-	nodeFields fields.Set,
-) bool {
+) (bool, error) {
+	var errors []error
```

Comment:
if we're accepting a *Node object, should we check for nil and return `false, nil` early?